File: Problems/No.2424_Longest_Uploaded_Prefix/longest_uploaded_prefix.py
class LUPrefix(object):

    def __init__(self, n):
        """
        :type n: int
        """
        self.Vstream = set()
        self.LUP = 0
        

    def upload(self, video):
        """
        :type video: int
        :rtype: None
        """
        self.Vstream.add(video)
        while self.LUP + 1 in self.Vstream:
            self.LUP += 1
        

    def longest(self):
        """
        :rtype: int
        """
        return self.LUP
        


# Your LUPrefix object will be instantiated and called as such:
# obj = LUPrefix(n)
# obj.upload(video)
# param_2 = obj.longest()

File: Problems/No.2424_Longest_Uploaded_Prefix/longest_uploaded_prefix.cpp
class LUPrefix {

private:
    int LUP = 0;
    std::set<int> Vstream;

public:
    LUPrefix(int n) {Vstream.insert(0);}
    
    void upload(int video) {

        Vstream.insert(video);
        while (Vstream.count(LUP+1)) LUP++;

    }
    
    int longest() {return LUP;}
};

/**
 * Your LUPrefix object will be instantiated and called as such:
 * LUPrefix* obj = new LUPrefix(n);
 * obj->upload(video);
 * int param_2 = obj->longest();
 */
